### PR TITLE
Document Screen Implementation

### PR DESCRIPTION
**Pull request by FlorianDinant**
## Summary

This PR introduces the DocumentScreen component, designed for displaying and downloading various document types. It supports multiple formats, including PDF, image formats (JPG, PNG, etc.), and SVG, allowing users to view the document and download it to their local device. The component also includes error handling and a page number display for PDFs.

### 1. Document Viewer Integration
•	Added support for multiple document formats (PDF, JPG, PNG, SVG, etc.).
•	Used react-native-pdf for PDF viewing and Image component for image formats.
•	Incorporated SVGImage for displaying SVG files.
•	Added a fallback view for unsupported document formats (GIF, WebP).

### 2. Document Download Functionality
•	Introduced a function to download the document and encode it in Base64.
•	Utilizes Expo’s FileSystem for file downloading and saving to a selected directory.
•	Handles errors related to file download and saving.

### 3. Error Handling and Feedback
•	Added error state to display appropriate error messages when document loading fails.
•	Used the ActivityIndicator to show loading state for PDF documents.

### 4. Page Number Display for PDFs
•	Display the current page number and total page count for PDF documents.
•	Shows page number in the top-left corner with dynamic updating during page navigation.

## Key Files Changed
•	`DocumentScreen.tsx`: Main screen component that handles document rendering, downloading, and displaying page numbers.
•	`SVGImage.tsx`: Component which allows to display a svg image from a url.
•	`i18config.ts`: Language configuration file for translation handling (e.g., error messages).

## Features
•	Document Formats Supported:
•	**PDF**: Rendered using react-native-pdf with activity indicator.
•	Images (**JPG**, **PNG**, **JPEG**): Rendered using Image component.
•	**SVG**: Rendered using a custom SVGImage component.
•	Unsupported formats (**GIF**, **WebP**): Displays an error message. BUT WILL BE AVAILABLE SOON (need to rebuild project)
•	Download & Save:
•	Allows users to download the document and save it locally using Expo’s FileSystem API.
•	Supports saving in selected directories with the appropriate permissions.

## Screenshots / Demos
Screenshot 2024-12-13 at 03 47 43
Screenshot 2024-12-13 at 03 46 40
Screenshot 2024-12-13 at 03 47 31
Screenshot 2024-12-13 at 03 48 06
Screenshot 2024-12-13 at 03 48 16

## Testing

1. Manual Testing
•	Verified document loading and error handling for all supported formats (PDF, Image, SVG).
•	Checked the download functionality to ensure the file is correctly saved in the selected directory.
•	Test edge cases behaviour

2. Automated Tests
•	run `edweiss-app/__test__/app/(app)/courses/[id]/materials/[materialId]/index.test.tsx`
•	run `edweiss-app/__test__/components/core/SVGImage.test.tsx`

## Notes
•	The current implementation supports basic error handling for document formats. However, additional edge cases can be tested further.
•	The download functionality requires directory permissions, which should be handled properly in a production environment.

The professor upload page will be implemented in my next PR

### CODE REVIEW

File: edweiss-app/components/core/SVGImage.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+const SVGImage: ReactComponent<SVGImageProps> = ({ uri, width, height, color, title, onError, onLoad, fallback }) => {
+
+    const [svgData, setSvgData] = useState(null);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        const fetchSvg = async () => {
+            try {
+                const response = await axios.get(uri);
```
Comment (holoisme) [inline edweiss-app/components/core/SVGImage.tsx:28]: Axios? Why add this dependency when you can use Firebase's Cloud Storage fetching functions (or JS `fetch`)? This is a big dependency to add considering it's only used here

**Review by holoisme — CHANGES_REQUESTED** [review]
This is a very good PR overall, I didn't have much to say about the quality of your code, it is also very well documented, fully localized, and has a very nice design overall !

My only confusion point would be the inclusion of the Axios dependency, which feels a bit out of place considering we're working with Firebase, maybe try to look up ways to replace the single call to axios with a Firebase function?

So yeah, keep it up, we're almost done :+1:
